File: HedgeHerd/functions/chat-completion.py
from flask import Flask, request, jsonify
from openai import OpenAI
from dotenv import load_dotenv
import os

load_dotenv()

app = Flask(__name__)
client = OpenAI(api_key="")

@app.route('/chat', methods=['POST'])
def chat():
    data = request.get_json()
    message = data['message']

    response = client.chat.completions.create(
        model="gpt-3.5-turbo",
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": message},
        ]
    )
    reply = response.choices[0].message.content
    return jsonify({'reply': reply})

if __name__ == '__main__':
    app.run(debug=True)

File: HedgeHerd/functions/chat-completion.js
const functions = require("firebase-functions");
const admin = require("firebase-admin");
const OpenAi = require("openai")

if (admin.apps.length === 0) {
    admin.initializeApp();
}

exports.chatCompletion = functions.https.onCall(async (data, context) => {
    const { prompt } = data;

    const openai = new OpenAi({
        apiKey: process.env.OPENAI_API_KEY
    });
    
    const aiModel = "gpt-4.1-nano"

    const messages = [
        {
            role:"system",
            content:"you are a helpful assistant"
        },
        {
            role:"user",
            content:prompt
        },
    ]

    const completion = await openai.chat.completions.create({
        model: aiModel,
        messages: messages
    })

    const aiResponse = completion.choices[0].message.content

    return {
        aiResponse
    }
})
